### ISSUE TITLE
oras pull doesn't pull content with oras artifacts

### ISSUE THREAD

**Issue by SteveLasker** [oras-project/oras#336]
- Initial setup

  ```bash
  ACR_NAME=stevelasprepro
  REGISTRY=$ACR_NAME.azurecr.io

  az group create -n $ACR_NAME -l southcentralus
  az acr create -n $ACR_NAME -g $ACR_NAME --zone-redundancy enabled --sku Premium
  az configure --default acr=$ACR_NAME
  az acr login 

  USERNAME='oras-token'
  PASSWORD=$(az acr token create -n $USERNAME \
                      -r $ACR_NAME \
                      --scope-map _repositories_admin \
                      --only-show-errors \
                      -o json | jq -r ".credentials.passwords[0].value")

  oras login -u $USERNAME -p $PASSWORD $REGISTRY

  REPO=net-monitor
  IMAGE_TAG=v1
  IMAGE=$REGISTRY/$REPO:$IMAGE_TAG
  ARTIFACT_TAG=regdoc-v1
  ARTIFACT=$REGISTRY/$REPO:$ARTIFACT_TAG
  ```

- Build and push an image

  ```bash
  docker build -t $IMAGE https://github.com/wabbit-networks/net-monitor.git#main
  docker push $IMAGE
  ```

- Create an independent artifact

  ```bash
  echo 'Readme Content' > readme.md
  echo 'Detailed Content' > readme-details.md
  ```

- Push the independent artifact

  ```bash
  oras push $ARTIFACT \
    --manifest-config /dev/null:application/vnd.oras.regdoc.config.v0+json \
    ./readme.md:application/markdown \
    ./readme-details.md:application/markdown
  ```

- Pull an independent artifact

  ```bash
  mkdir download
  cd download
  oras pull $ARTIFACT -a
  cd ../
  ```

- Content exists
  ```bash
  ls
  readme-details.md  readme.md
  ```

## ORAS Artifact - Reference Types

- Push the regDoc to the registry, as a reference to the $IMAGE:

  ```bash
  oras push $REGISTRY/$REPO \
      --artifact-type 'regdoc/example' \
      --subject $IMAGE \
    ./readme.md:application/markdown \
    ./readme-details.md:application/markdown
  ```

- Get the digest of the regdoc sample
  ```bash
  REGDOC_DIGEST=$(oras discover -o json \
                    --artifact-type 'regdoc/example' \
                    -u $USERNAME -p $PASSWORD \
                    $IMAGE | jq -r ".references[0].digest")
  ```

- Pull the reference type content

  ```bash
  cd download
  oras pull -a $REGISTRY/$REPO@$REGDOC_DIGEST
    ```

- Content exists
  ```bash
  ls
  # EMPTY
  ```

## Expected Behavior

Contents pulled

**Comment by shizhMSFT** [comment]
@SteveLasker The branch [artifacts](https://github.com/oras-project/oras/tree/artifacts) works as expected.

Note to `zsh` users: setting `USERNAME` environment variable has no effect. Please use other environment names.

**Comment by SteveLasker** [comment]
Confirmed the artifacts branch is working. @shizhMSFT will cut a new release for folks to download.
@shizhMSFT, please tag this issue when the release is done and we'll close this issue.

**Comment by shizhMSFT** [comment]
Closing this issue as it has been addressed by [v0.2.1-alpha.1](https://github.com/oras-project/oras/releases/tag/v0.2.1-alpha.1)
